nomad/resource_job_v2.go

```diff
@@ -0,0 +1,1682 @@
+			}
+		}
+
+		templates := make([]*api.Template, 0)
+		for _, tpl := range t["template"].([]interface{}) {
+			tp := tpl.(map[string]interface{})
+
+			template := &api.Template{
+				ChangeMode:   getString(tp, "change_mode"),
+				ChangeSignal: getString(tp, "change_signal"),
+				EmbeddedTmpl: getString(tp, "data"),
+				DestPath:     getString(tp, "destination"),
+				Envvars:      getBool(tp, "env"),
+				LeftDelim:    getString(tp, "left_delimiter"),
+				Perms:        getString(tp, "perms"),
+				RightDelim:   getString(tp, "right_delimiter"),
+				SourcePath:   getString(tp, "source"),
+			}
+
+			duration, err := getDuration(tp["splay"])
+			if err != nil {
+				return nil, err
+			}
+			template.Splay = duration
+
+			duration, err = getDuration(tp["vault_grace"])
+			if err != nil {
+				return nil, err
+			}
+			template.VaultGrace = duration
+
+			templates = append(templates, template)
+		}
+
+		volumeMounts := make([]*api.VolumeMount, 0)
+		for _, vm := range t["volume_mount"].([]interface{}) {
+			v := vm.(map[string]interface{})
+
+			volumeMount := &api.VolumeMount{
+				Volume:      getString(v, "volume"),
+				Destination: getString(v, "destination"),
+				ReadOnly:    getBool(v, "read_only"),
+			}
+
+			volumeMounts = append(volumeMounts, volumeMount)
+		}
+
+		var config map[string]interface{}
+		err := json.Unmarshal([]byte(t["config"].(string)), &config)
+		if err != nil {
+			return nil, err
+		}
+
+		services, err := getServices(t["service"])
+		if err != nil {
+			return nil, err
+		}
+
+		task := &api.Task{
+			Name:            t["name"].(string),
+			Config:          config,
+			Meta:            getMapOfString(t["meta"]),
+			Driver:          t["driver"].(string),
+			KillSignal:      t["kill_signal"].(string),
+			Leader:          t["leader"].(bool),
+			User:            t["user"].(string),
+			Kind:            t["kind"].(string),
+			Artifacts:       artifacts,
+			Constraints:     getConstraints(t["constraint"]),
+			Affinities:      getAffinities(t["affinity"]),
+			DispatchPayload: dispatchPayloadConfig,
+			Env:             getMapOfString(t["env"]),
+			Lifecycle:       taskLifecycle,
+			LogConfig:       getLogConfig(t["logs"]),
+			Resources:       getResources(t["resources"]),
+			Services:        services,
+			Templates:       templates,
+			Vault:           getVault(t["vault"]),
+			VolumeMounts:    volumeMounts,
+		}
+
+		duration, err := getDuration(t["kill_timeout"])
+		if err != nil {
+			return nil, err
+		}
+		task.KillTimeout = duration
+
+		duration, err = getDuration(t["shutdown_delay"])
+		if err != nil {
+			return nil, err
+		}
+		if duration != nil {
+			task.ShutdownDelay = *duration
+		}
+
+		tasks = append(tasks, task)
+	}
+
+	return tasks, nil
+}
+
+func getNetworks(d interface{}) []*api.NetworkResource {
+	networks := make([]*api.NetworkResource, 0)
+
+	for _, nr := range d.([]interface{}) {
+		n := nr.(map[string]interface{})
+
+		// TODO(remi): find what to do with the ports here
```

Comment:
The [`parsePorts` function checks if there's a Value set](https://github.com/hashicorp/nomad/blob/master/jobspec/parse_network.go#L105-L109) (where [`Value` maps to `static`](https://github.com/hashicorp/nomad/blob/master/api/resources.go#L88) in the jobspec).

I think that's what you would need to do here as well.

Reply:
Thanks!